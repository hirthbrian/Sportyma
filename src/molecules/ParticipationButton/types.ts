import { ImageSourcePropType, GestureResponderEvent } from "react-native";

export interface Props {
  title: string;
  isActive?: boolean;
  activeColor?: string;
  icon?: ImageSourcePropType;
  onPress: (event: GestureResponderEvent) => void;
}
