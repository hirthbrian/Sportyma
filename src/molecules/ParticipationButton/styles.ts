import styled from "styled-components/native";

import { KanitText } from "../../atoms";

interface ContainerProps {
  color: string;
  isActive: boolean;
}

export const PressableContainer = styled.Pressable`
  flex: 1;
`;

export const Container = styled.View`
  height: 50px;
  background-color: ${(props: ContainerProps) =>
    props.isActive ? props.color : "#FFFFFF"};
  border-radius: 50px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  border-color: ${(props: ContainerProps) =>
    props.isActive ? props.color : "#2559f233"};
  border-width: 2px;
`;

export const Text = styled(KanitText)`
  letter-spacing: 1px;
  color: ${(props: ContainerProps) => (props.isActive ? "#FFFFFF" : "#2559F2")};
`;

export const Image = styled.Image`
  width: 18px;
  height: 18px;
  margin-right: 10px;
  tint-color: ${(props: ContainerProps) =>
    props.isActive ? "#FFFFFF" : "#2559F2"};
`;
