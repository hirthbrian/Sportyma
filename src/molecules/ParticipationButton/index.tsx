import Animated, {
  withTiming,
  useSharedValue,
  useAnimatedStyle,
} from "react-native-reanimated";

import { Props } from "./types";
import { PressableContainer, Container, Text, Image } from "./styles";

const ParticipationButton = ({
  title,
  isActive,
  activeColor,
  icon,
  onPress,
}: Props) => {
  const scale = useSharedValue(1);

  const animatedStyle = useAnimatedStyle(() => ({
    transform: [{ scale: withTiming(scale.value, { duration: 200 }) }],
  }));

  const onPressIn = () => (scale.value = 0.95);
  const onPressOut = () => (scale.value = 1);

  return (
    <PressableContainer
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      onPress={onPress}
    >
      <Animated.View style={animatedStyle}>
        <Container isActive={isActive} color={activeColor}>
          <Image source={icon} isActive={isActive} color={activeColor} />
          <Text size={13} weight={500} isActive={isActive}>
            {title.toUpperCase()}
          </Text>
        </Container>
      </Animated.View>
    </PressableContainer>
  );
};

export default ParticipationButton;
