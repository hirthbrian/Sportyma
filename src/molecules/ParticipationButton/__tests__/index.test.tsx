import React from "react";
import renderer from "react-test-renderer";

import ParticipationButton from "..";

describe("<ParticipationButton />", () => {
  let props;

  beforeEach(() => {
    props = {
      title: "test",
      isActive: false,
      activeColor: "red",
    };
  });

  it("renders correcly", () => {
    expect(
      renderer.create(<ParticipationButton {...props} />).toJSON()
    ).toBeTruthy();
  });
  it("match snapshot", () => {
    const tree = renderer.create(<ParticipationButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("match snapshot with active", () => {
    props = {
      ...props,
      isActive: true,
    };
    const tree = renderer.create(<ParticipationButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("match snapshot with color", () => {
    props = {
      ...props,
      activeColor: "blue",
    };
    const tree = renderer.create(<ParticipationButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
