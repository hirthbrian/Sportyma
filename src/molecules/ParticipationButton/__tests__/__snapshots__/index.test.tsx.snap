// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipationButton /> match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      color="red"
      isActive={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#2559f233",
            "borderRadius": 50,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        color="red"
        isActive={false}
        style={
          Array [
            Object {
              "height": 18,
              "marginRight": 10,
              "tintColor": "#2559F2",
              "width": 18,
            },
          ]
        }
      />
      <Text
        isActive={false}
        size={13}
        style={
          Array [
            Object {
              "color": "#2559F2",
              "fontFamily": "Kanit-Medium",
              "fontSize": 13,
              "letterSpacing": 1,
            },
          ]
        }
        weight={500}
      >
        TEST
      </Text>
    </View>
  </View>
</View>
`;

exports[`<ParticipationButton /> match snapshot with active 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      color="red"
      isActive={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderColor": "red",
            "borderRadius": 50,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        color="red"
        isActive={true}
        style={
          Array [
            Object {
              "height": 18,
              "marginRight": 10,
              "tintColor": "#FFFFFF",
              "width": 18,
            },
          ]
        }
      />
      <Text
        isActive={true}
        size={13}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Kanit-Medium",
              "fontSize": 13,
              "letterSpacing": 1,
            },
          ]
        }
        weight={500}
      >
        TEST
      </Text>
    </View>
  </View>
</View>
`;

exports[`<ParticipationButton /> match snapshot with color 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      color="blue"
      isActive={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#2559f233",
            "borderRadius": 50,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        color="blue"
        isActive={false}
        style={
          Array [
            Object {
              "height": 18,
              "marginRight": 10,
              "tintColor": "#2559F2",
              "width": 18,
            },
          ]
        }
      />
      <Text
        isActive={false}
        size={13}
        style={
          Array [
            Object {
              "color": "#2559F2",
              "fontFamily": "Kanit-Medium",
              "fontSize": 13,
              "letterSpacing": 1,
            },
          ]
        }
        weight={500}
      >
        TEST
      </Text>
    </View>
  </View>
</View>
`;
